rm-datasource/src/main/java/io/seata/rm/datasource/AsyncWorker.java

```diff
@@ -59,8 +60,125 @@
 
     private static final int UNDOLOG_DELETE_LIMIT_SIZE = 1000;
 
+    private static final int ASYNC_COMMIT_BUFFER_LIMIT = ConfigurationFactory.getInstance().getInt(
+        CLIENT_ASYNC_COMMIT_BUFFER_LIMIT, DEFAULT_CLIENT_ASYNC_COMMIT_BUFFER_LIMIT);
+
+    private final BlockingQueue<Phase2Context> commitQueue;
+
+    private final ScheduledExecutorService scheduledExecutor;
+
+    public AsyncWorker() {
+        LOGGER.info("Async Commit Buffer Limit: {}", ASYNC_COMMIT_BUFFER_LIMIT);
+        commitQueue = new LinkedBlockingQueue<>(ASYNC_COMMIT_BUFFER_LIMIT);
+
+        ThreadFactory threadFactory = new NamedThreadFactory("AsyncWorker", 2, true);
+        scheduledExecutor = new ScheduledThreadPoolExecutor(2, threadFactory);
+        scheduledExecutor.scheduleAtFixedRate(this::doBranchCommits, 10, 1000, TimeUnit.MILLISECONDS);
```

Comment:
may be throws thread interrupt exception or Error. suggest try catch Throwable.

Reply:
fixed